pivotx/lib.php

```diff
@@ -4453,15 +4508,25 @@ function writeFile($filename, $output, $mode='w') {
         return;
     }
 
+    if( !flock( $fh, LOCK_EX  ) ){
+        _e('Lock Error. Could not lock file for writing').": ".$filename."<br />\n";
+        fclose( $fh );    
+        return;
+    }
     // wrrrriting!
     if(!fwrite($fh, $output)) {
```

Comment:
This is probably the most crucial part and will fix index db corruption, this write needs to protected by a mutex

Reply:
FYI: The function `writeFile` is very little used and is not the cause of the entry index corruption. For reasons unknown to me this function is almost exclusively used to writing tag data.  It's the function `saveSerialize` that already had the lock protection, that does the important work. Anyway, as you explained in the issue - `loadSerialize` needs lock protection too since the file is temporarily emptied in `saveSerialize`.

Reply:
I only have partial recollection of the issue but it makes sens. 

I also vaguely remember that a lot of operations only involving viewing pages would trigger a call to `saveSerialize` for no apparent reasons.

Reply:
Yes, that was why I wrote "as you explained in the issue" ;-)